crates/oxc_query/src/properties.rs

```diff
@@ -0,0 +1,584 @@
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_class_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "extended_class_name" => resolve_property_with(contexts, |v| {
+            let Some(Expression::Identifier(ref ident)) = v
+                .as_class()
+                .unwrap_or_else(|| panic!("expected to have a class vertex, instead have: {v:#?}"))
+                .class
+                .super_class
+            else {
+                return FieldValue::Null;
+            };
+            ident.name.to_string().into()
+        }),
+        "is_abstract" => resolve_property_with(contexts, |v| {
+            v.as_class()
+                .unwrap_or_else(|| panic!("expected to have a class vertex, instead have: {v:#?}"))
+                .class
+                .modifiers
+                .contains(oxc_ast::ast::ModifierKind::Abstract)
+                .into()
+        }),
+        _ => {
+            unreachable!("attempted to read unexpected property '{property_name}' on type 'Class'")
+        }
+    }
+}
+
+pub(super) fn resolve_class_method_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "accessibility" => resolve_property_with(contexts, |v| {
+            v.as_class_method()
+                .unwrap_or_else(|| {
+                    panic!("expected to have a classmethod vertex, instead have: {v:#?}")
+                })
+                .method
+                .accessibility
+                .map(accessibility_to_string)
+                .map_or(FieldValue::Null, Into::into)
+        }),
+        "is_abstract" => resolve_property_with(contexts, |v| {
+            v.as_class_method()
+                .unwrap_or_else(|| {
+                    panic!("expected to have a classmethod vertex, instead have: {v:#?}")
+                })
+                .is_abstract
+                .into()
+        }),
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'ClassMethod'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_class_property_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "accessibility" => resolve_property_with(contexts, |v| {
+            v.as_class_property()
+                .unwrap_or_else(|| {
+                    panic!("expected to have a classproperty vertex, instead have: {v:#?}")
+                })
+                .property
+                .accessibility
```

Comment:
I wonder if accessibility should be a trait method? (that's not in oxc_query though)

Reply:
Let's not